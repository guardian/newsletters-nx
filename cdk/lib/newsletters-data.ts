import type { GuStackProps } from '@guardian/cdk/lib/constructs/core';
import { GuStack } from '@guardian/cdk/lib/constructs/core';
import { GuS3Bucket } from "@guardian/cdk/lib/constructs/s3";
import type { App } from 'aws-cdk-lib';
import { StringParameter } from "aws-cdk-lib/aws-ssm"

export class NewslettersData extends GuStack {
	constructor(scope: App, id: string, props: GuStackProps) {
		super(scope, id, props);

		const app = 'newsletters-data'
		const bucketParameterName = `/${this.stage}/${this.stack}/${app}/s3BucketName`
		const bucketName = StringParameter.valueForStringParameter(this, bucketParameterName)

		new GuS3Bucket(this, "DataBucket", {
      bucketName,
      app,
      versioned: true,
    });
	}
}
