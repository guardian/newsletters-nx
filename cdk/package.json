{
  "name": "cdk",
  "version": "0.0.0",
  "private": true,
  "scripts": {
    "build": "tsc",
    "test": "jest",
    "test-update": "jest -u",
    "format": "prettier --write \"{lib,bin}/**/*.ts\"",
    "lint": "eslint lib/** bin/** --ext .ts --no-error-on-unmatched-pattern",
    "synth": "cdk synth --path-metadata false --version-reporting false",
    "diff": "cdk diff --path-metadata false --version-reporting false"
  },
  "devDependencies": {
    "@guardian/cdk": "49.0.3",
    "@guardian/eslint-config-typescript": "3.0.0",
    "@guardian/prettier": "1.0.0",
    "@types/jest": "^29.2.5",
    "@types/node": "18.11.18",
    "aws-cdk": "2.59.0",
    "aws-cdk-lib": "2.59.0",
    "constructs": "10.1.215",
    "eslint": "^8.32.0",
    "jest": "^29.3.1",
    "prettier": "^2.8.3",
    "source-map-support": "^0.5.20",
    "ts-jest": "^29.0.5",
    "ts-node": "^10.9.1",
    "typescript": "~4.9.4"
  },
  "prettier": "@guardian/prettier",
  "jest": {
    "testMatch": [
      "<rootDir>/lib/**/*.test.ts"
    ],
    "transform": {
      "^.+\\.tsx?$": "ts-jest"
    },
    "setupFilesAfterEnv": [
      "./jest.setup.js"
    ]
  },
  "eslintConfig": {
    "root": true,
    "env": {
      "node": true,
      "jest": true
    },
    "extends": [
      "@guardian/eslint-config-typescript"
    ],
    "parserOptions": {
      "ecmaVersion": 2020,
      "sourceType": "module"
    },
    "plugins": [
      "@typescript-eslint"
    ],
    "rules": {
      "@typescript-eslint/no-inferrable-types": 0,
      "import/no-namespace": 2
    },
    "ignorePatterns": [
      "**/*.js",
      "node_modules",
      "cdk.out",
      ".eslintrc.js",
      "jest.config.js"
    ]
  }
}