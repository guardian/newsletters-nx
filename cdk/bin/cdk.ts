import 'source-map-support/register';
import { App } from 'aws-cdk-lib';
import { NewslettersData } from '../lib/newsletters-data';

const app = new App();

export const newslettersProps = {
	stack: 'newsletters',
	env: {
		account: 'frontend',
		region: 'eu-west-1'
	}
}

new NewslettersData(app, 'NewslettersData-DEV', {
	...newslettersProps,
	stage: 'DEV',
});
