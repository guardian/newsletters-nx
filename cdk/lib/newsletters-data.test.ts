import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { newslettersProps } from '../bin/cdk';
import { NewslettersData } from './newsletters-data';

describe('The Newsletters Data stack', () => {
	it('matches the snapshot', () => {
		const app = new App();
		const stack = new NewslettersData(app, 'NewslettersData-TEST', {
			...newslettersProps,
			stage: 'TEST',
		});
		const template = Template.fromStack(stack);
		expect(template.toJSON()).toMatchSnapshot();
	});
});
