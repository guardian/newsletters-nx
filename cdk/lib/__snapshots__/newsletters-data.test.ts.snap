// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Newsletters API stack matches the snapshot 1`] = `
{
  "Metadata": {
    "gu:cdk:constructs": [
      "GuS3Bucket",
    ],
    "gu:cdk:version": "TEST",
  },
  "Parameters": {
    "SsmParameterValueTESTnewslettersnewslettersdatas3BucketNameC96584B6F00A464EAD1953AFF4B05118Parameter": {
      "Default": "/TEST/newsletters/newsletters-data/s3BucketName",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "DataBucketNewslettersdata573FDBE4": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "BucketName": {
          "Ref": "SsmParameterValueTESTnewslettersnewslettersdatas3BucketNameC96584B6F00A464EAD1953AFF4B05118Parameter",
        },
        "PublicAccessBlockConfiguration": {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
        "Tags": [
          {
            "Key": "App",
            "Value": "newsletters-data",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/newsletters-nx",
          },
          {
            "Key": "Stack",
            "Value": "newsletters",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "VersioningConfiguration": {
          "Status": "Enabled",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
  },
}
`;
